packages/react/src/components/Comments/Comments.tsx

```diff
@@ -3,45 +3,75 @@ import { ThreadData } from "@blocknote/core/comments";
 import { useComponentsContext } from "../../editor/ComponentsContext.js";
 import { useDictionary } from "../../i18n/dictionary.js";
 import { Comment } from "./Comment.js";
+import { useBlockNoteEditor } from "../../hooks/useBlockNoteEditor.js";
+import { useUsers } from "./useUsers.js";
 
 export type CommentsProps = {
   thread: ThreadData;
-  collapse?: boolean;
+  maxCommentsBeforeCollapse?: number;
 };
 
-export const Comments = ({ thread, collapse }: CommentsProps) => {
+export const Comments = ({
+  thread,
+  maxCommentsBeforeCollapse,
+}: CommentsProps) => {
   const Components = useComponentsContext()!;
   const dict = useDictionary();
+  const editor = useBlockNoteEditor();
+  const users = useUsers(editor, [thread.resolvedBy!]);
 
-  if (collapse) {
-    return [
-      <Comment
-        key={thread.comments[0].id}
-        thread={thread}
-        comment={thread.comments[0]}
-        showResolveButton={true}
-      />,
+  // Maps all comments to elements.
+  const comments = thread.comments.map((comment, index) => (
+    <Comment
+      key={comment.id}
+      thread={thread}
+      comment={comment}
+      showResolveButton={index === 0}
+    />
+  ));
+
+  // Adds "resolved by" comment if needed.
+  if (thread.resolved && thread.resolvedUpdatedAt) {
+    const resolvedCommentIndex =
+      thread.comments.findLastIndex(
+        (comment) =>
+          thread.resolvedUpdatedAt!.getTime() > comment.createdAt.getTime()
+      ) + 1;
+
+    comments.splice(
+      resolvedCommentIndex,
+      0,
+      <Components.Comments.Comment
+        className={"bn-thread-comment"}
+        authorInfo={users.get(thread.resolvedBy!) || "loading"}
+        timeString={thread.resolvedUpdatedAt!.toLocaleDateString(undefined, {
+          month: "short",
+          day: "numeric",
+        })}
+        edited={false}
+        showActions={false}>
+        <div className={"bn-resolved-text"}>
+          {dict.comments.sidebar.marked_as_resolved}
+        </div>
+      </Components.Comments.Comment>
+    );
+  }
+
+  // Collapses replies if needed.
+  if (
```

Comment:
could this collapse the "marked as resolved comment"? how does that work in gdocs?

Reply:
GDocs treats the "Marked as resolved" comments the same way as regular comments when it comes to collapsing threads. There is the caveat that GDocs doesn't let you reply to resolved threads (it will reopen them if you do), which we can quite easily change if needed. GitHub though, for example, does let you respond to resolved threads.